File: api/tags.py
import services.tag_service as service
from api.responses import safe


def get_tags(user_id):
    return safe(lambda: service.get_tags(user_id))


def get_tag(tag_id):
    return safe(lambda: service.get_tag(tag_id))


def create_tag(user_id, name):
    return safe(lambda: service.create_tag(user_id, name))


def update_tag(tag_id, name=None):
    return safe(lambda: service.update_tag(tag_id, name=name))


def delete_tag(tag_id):
    return safe(lambda: service.delete_tag(tag_id))


def get_transaction_tags(transaction_id):
    return safe(lambda: service.get_transaction_tags(transaction_id))


def add_tag_to_transaction(transaction_id, tag_id):
    return safe(lambda: service.add_tag_to_transaction(transaction_id, tag_id))


def remove_tag_from_transaction(transaction_id, tag_id):
    return safe(lambda: service.remove_tag_from_transaction(transaction_id, tag_id))

File: frontend/src/lib/api/tags.ts
import { callApi } from './client';

export type Tag = {
	tag_id: number;
	user_id: number;
	name: string;
};

export function getTags(userId: number) {
	return callApi<Tag[]>('get_tags', userId);
}

export function getTag(tagId: number) {
	return callApi<Tag | null>('get_tag', tagId);
}

export function createTag(payload: { user_id: number; name: string }) {
	return callApi<{ tag_id: number }>('create_tag', payload.user_id, payload.name);
}

export function updateTag(tagId: number, payload: { name?: string | null } = {}) {
	return callApi<{ status: string }>('update_tag', tagId, payload.name ?? null);
}

export function deleteTag(tagId: number) {
	return callApi<{ status: string }>('delete_tag', tagId);
}

export function getTransactionTags(transactionId: number) {
	return callApi<Tag[]>('get_transaction_tags', transactionId);
}

export function addTagToTransaction(transactionId: number, tagId: number) {
	return callApi<{ status: string }>('add_tag_to_transaction', transactionId, tagId);
}

export function removeTagFromTransaction(transactionId: number, tagId: number) {
	return callApi<{ status: string }>('remove_tag_from_transaction', transactionId, tagId);
}
